### PR TITLE
Feat/fingerprint

### PR DESCRIPTION
**Pull request by oaliaga**
+ Add fingerprint verification.
+ Use of location is optimized.

### CODE REVIEW

File: app/src/main/java/com/prey/activities/CheckPasswordHtmlActivity.java [hunk 1]
```diff
@@ -549,4 +577,139 @@ public void askForPermissionNotification() {
...
+        keyStore.load(null);
+        if (keyStore.containsAlias(keyName)) {
+            PublicKey publicKey = keyStore.getCertificate(keyName).getPublicKey();
+            PrivateKey privateKey = (PrivateKey) keyStore.getKey(keyName, null);
+            return new KeyPair(publicKey, privateKey);
+        }
+        return null;
+    }
+
+    public void abrir(){
```
Comment (patriciojofre) [inline app/src/main/java/com/prey/activities/CheckPasswordHtmlActivity.java:697]: english

File: app/src/main/java/com/prey/events/factories/EventFactory.java [hunk 2]
```diff
@@ -218,11 +218,10 @@ public static boolean isValidLowBattery(Context ctx) {
      * @return if you have all permissions
      */
     public static boolean verifyNotification(Context ctx) {
-        boolean canAccessCamera = PreyPermission.canAccessCamera(ctx);
         boolean canAccessCoarseLocation = PreyPermission.canAccessCoarseLocation(ctx);
         boolean canAccessFineLocation = PreyPermission.canAccessFineLocation(ctx);
         boolean canAccessStorage = PreyPermission.canAccessStorage(ctx);
-        return canAccessCamera && (canAccessCoarseLocation || canAccessFineLocation) && canAccessStorage;
+        return (canAccessCoarseLocation || canAccessFineLocation) && canAccessStorage;
```
Comment (patriciojofre) [inline app/src/main/java/com/prey/events/factories/EventFactory.java:224]: ya no va el de camara?

File: app/src/main/java/com/prey/activities/CheckPasswordHtmlActivity.java [hunk 3]
```diff
@@ -549,4 +577,139 @@ public void askForPermissionNotification() {
...
+        if (PreyPermission.checkBiometricSupport(this)) {  // Check whether this device can authenticate with biometrics
+            Signature signature;
+            try {
+                KeyPair keyPair =  generateKeyPair(KEY_NAME, true);
+                mToBeSignedMessage = Base64.encodeToString(keyPair.getPublic().getEncoded(), Base64.URL_SAFE) +
+                        ":" +
+                        KEY_NAME +
+                        ":" +
+                        // Generated by the server to protect against replay attack
+                        "12345";
```
[the remark this answers is not in the document]
Comment (oaliaga) [reply app/src/main/java/com/prey/activities/CheckPasswordHtmlActivity.java]: Se quito

File: app/src/main/java/com/prey/activities/CheckPasswordHtmlActivity.java (hunk 1 above)
Comment (oaliaga) [reply app/src/main/java/com/prey/activities/CheckPasswordHtmlActivity.java]: Se cambio a openBiometric

**Review by patriciojofre — APPROVED** [review]

**Pull request merged**